sdcm/nemesis.py

```diff
@@ -4413,6 +4419,90 @@ def disrupt_add_remove_mv(self):
         finally:
             self.unset_current_running_nemesis(cql_query_executor_node)
 
+    def disrupt_toggle_audit(self):
+        """
+            Enable audit log with all categories and user keyspaces (if audit already enabled, disable it and finish the Nemesis),
+            verify audit log content,
+            reduce categories by excluding DML and QUERY,
+            verify DDL are logged in audit log correctly. Leaves audit log enabled this way.
+        """
+        if not self.target_node.is_enterprise:
+            raise UnsupportedNemesis("Auditing feature is only supported by Scylla Enterprise")
+
+        if not (test_ks_cf_list := self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)):
+            raise UnsupportedNemesis("Not found non-system keyspaces")
+
+        audit = Audit(self.cluster)
+
+        if audit.is_enabled():
+            InfoEvent("Audit was enabled -> disabling it").publish()
+            audit.disable()
+            return
+
+        # audit_keyspace = "audit_keyspace"
+        # with self.cluster.cql_connection_patient(node=self.target_node) as session:
+        #     query = f"CREATE KEYSPACE IF NOT EXISTS {audit_keyspace} with replication = " \
+        #             f"{{'class': 'NetworkTopologyStrategy', 'replication_factor': 3}}"
+        #     session.execute(query)
+
+        keyspaces_for_audit = [ks_cf.split(".")[0] for ks_cf in test_ks_cf_list]
+        # keyspaces_for_audit = [audit_keyspace]
+        InfoEvent(f"Enabling full audit for keyspaces: {keyspaces_for_audit}").publish()
+        audit_config = AuditConfiguration(
+            store="table",
+            categories=["DCL", "DDL", "AUTH", "ADMIN", "DML", "QUERY"],
+            keyspaces=keyspaces_for_audit,
+            tables=[],
+        )
+        try:
+            audit.configure(audit_config)
+            keyspace_name = keyspaces_for_audit[0]
+            errors = []
+            audit_start = time.time()
+            time.sleep(300)  # Wait for audit log work for some time and not breaking Scylla
```

Comment:
I think 5 min it's so big. We can have the troubles to read from the audit table

Reply:
Yes, we found issues when having Audit with QUERY/DML on high traffic keyspaces anyway - redesigned the test to use separate ks only for Audit verification purposes.